src/tamp_improv/approaches/improvisational/policies/rl2mpc.py

```diff
@@ -0,0 +1,196 @@
+"""RL2MPC policy implementation combining RL and MPC."""
+
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import cast
+
+import gymnasium as gym
+import numpy as np
+
+from tamp_improv.approaches.improvisational.policies.base import (
+    ActType,
+    ObsType,
+    Policy,
+    PolicyContext,
+    TrainingData,
+)
+from tamp_improv.approaches.improvisational.policies.mpc import MPCConfig, MPCPolicy
+from tamp_improv.approaches.improvisational.policies.rl import (
+    RLConfig,
+    RLPolicy,
+    TrainingProgressCallback,
+)
+
+
+@dataclass
+class RL2MPCConfig:
+    """Configuration for RL2MPC policy."""
+
+    rl_config: RLConfig = field(default_factory=RLConfig)
+    mpc_config: MPCConfig = field(default_factory=MPCConfig)
+    reward_threshold: float = -5.0  # Threshold to switch from RL to MPC
+    window_size: int = 50  # Window size for computing average reward
+
+
+class RL2MPCCallback(TrainingProgressCallback):
+    """Callback that stops training when reward threshold is reached."""
```

Comment:
so the hope is that this should be faster to train than RL alone, right? if training time is now one of the metrics that we care about, can we add it to the final results table and verify that RL2MPC is better than RL in this way?